feat(rds): add Aurora MySQL versions 2.12.4, 3.08.2

Tietew opened the pull request:
### Reason for this change

New Aurora MySQL versions are available.
- 2.12.4 - [Database engine updates for Amazon Aurora MySQL version 2](https://docs.aws.amazon.com/AmazonRDS/latest/AuroraMySQLReleaseNotes/AuroraMySQL.Updates.20Updates.html)
- 3.08.2 - [Database engine updates for Amazon Aurora MySQL version 3](https://docs.aws.amazon.com/AmazonRDS/latest/AuroraMySQLReleaseNotes/AuroraMySQL.Updates.30Updates.html)

### Description of changes

Added above Aurora MySQL versions.

### Describe any new or updated permissions being added

N/A

### Description of how you validated changes

``` console
$ aws --region us-east-1 rds describe-db-engine-versions --engine aurora-mysql --query DBEngineVersions[].EngineVersion
[
    "5.7.mysql_aurora.2.11.1",
    "5.7.mysql_aurora.2.11.2",
    "5.7.mysql_aurora.2.11.3",
    "5.7.mysql_aurora.2.11.4",
    "5.7.mysql_aurora.2.11.5",
    "5.7.mysql_aurora.2.11.6",
    "5.7.mysql_aurora.2.12.0",
    "5.7.mysql_aurora.2.12.1",
    "5.7.mysql_aurora.2.12.2",
    "5.7.mysql_aurora.2.12.3",
    "5.7.mysql_aurora.2.12.4",
    "8.0.mysql_aurora.3.04.0",
    "8.0.mysql_aurora.3.04.1",
    "8.0.mysql_aurora.3.04.2",
    "8.0.mysql_aurora.3.04.3",
    "8.0.mysql_aurora.3.05.2",
    "8.0.mysql_aurora.3.06.0",
    "8.0.mysql_aurora.3.06.1",
    "8.0.mysql_aurora.3.07.0",
    "8.0.mysql_aurora.3.07.1",
    "8.0.mysql_aurora.3.08.0",
    "8.0.mysql_aurora.3.08.1",
    "8.0.mysql_aurora.3.08.2"
]
```

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

aws-cdk-automation reviewed (dismissed):
(This review is outdated)

Tietew commented:
Exemption Request: README and tests are not required generally when adding new engine versions.

shikha372 reviewed (changes requested):
Unit tests to be updated
[Tietew/aws-cdk@main/packages/aws-cdk-lib/aws-rds/test/cluster-engine.test.ts#L51](https://github.com/Tietew/aws-cdk/blob/main/packages/aws-cdk-lib/aws-rds/test/cluster-engine.test.ts#L51)
[Tietew/aws-cdk@main/packages/aws-cdk-lib/aws-rds/test/cluster-engine.test.ts#L170](https://github.com/Tietew/aws-cdk/blob/main/packages/aws-cdk-lib/aws-rds/test/cluster-engine.test.ts#L170)

Tietew commented:
@shikha372 I've updated tests.

Deleted tests in L.51-88 because they were duplicated.
Split a test in L.123-191 per engine because it was a bit long and the description was wrong.

Exemption for README is still needed.

Tietew commented:
Release note 3.08.2 is now available.
https://docs.aws.amazon.com/AmazonRDS/latest/AuroraMySQLReleaseNotes/AuroraMySQL.Updates.3082.html

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 545e5ff06122b6c6044f11c2804b269ca8ca86ff
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=095d3bfd-1f93-44d7-bc37-126bf898a1e3%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: d59c42244dfde06668802f76c46c11cc24ff096b
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=2d014952-f56b-45b9-b508-26773d7b9a6e%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: bbdcfcfce46e81d6cd2a69d3956f3c027710c60d
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=bf97f054-940c-4f07-a346-e93397c6516c%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->
